include/bluetooth/bluetooth.h

```diff
@@ -71,6 +79,56 @@ int bt_enable(bt_ready_cb_t cb);
  */
 int bt_set_id_addr(const bt_addr_le_t *addr);
 
+/** @brief Get the currently configured identities.
+ *
+ *  Returns an array of the currently configured identity addresses. To
+ *  make sure all available identities can be retreived, the number of
+ *  elements in the @a addrs array should be CONFIG_BT_ID_MAX.
+ *
+ *  @param addrs Array where to store the configured identities.
+ *  @param count Should be initialized to the array size. Once the function
+ *               returns it will contain the number of returned identies.
+ */
+void bt_id_get(bt_addr_le_t *addrs, size_t *count);
+
+/** @brief Create a new identity.
+ *
+ *  Create a new identity using the given address and IRK. This function
+ *  can be called before calling bt_enable(), in which case it can be used
+ *  to override the controller's public address (in case it has one). However,
+ *  the new identity will only be stored persistently in flash when this API
+ *  is used after bt_enable(). The reason is that the persistent settings
+ *  are loaded after bt_enable() and would therefore cause potential conflicts
+ *  with the stack blindly overwriting what's stored in flash. The identity
+ *  will also not be written to flash in case a pre-defined address is
+ *  provided, since in such a situation the app clearly has some place it got
+ *  the address from and will be able to repeat the procedure on every power
+ *  cycle, i.e. it would be redundant to also store the information in flash.
+ *
+ *  If the application wants to have the stack randomly generate identities
+ *  and store them in flash for later recovery, the way to do it would be
+ *  to first initialize the stack (using bt_enable), then call settings_load(),
+ *  and after that check with bt_id_get() how many identities were recovered.
+ *  If an insufficient amount of identities were recovered the app may then
+ *  call bt_id_create() to create new ones.
+ *
+ *  @param addr Address to use for the new identity. If initialized to
```

Comment:
I think it's less confusing if generating a random value looks the same for addr and irk. So NULL addr should generate a random value, and there could be a BT_IRK_LE_ANY defined to all zeroes.

Reply:
Yep, makes sense. I'll fix this in the next revision.

Reply:
Actually, doing this for the IRK seems a bit excessive. The documentation asks to use all zeroes, which is fairly straight-forward to do in the app, i.e. `u8_t irk[16] = { 0 };`. If I went with a define similar to BT_ADDR_LE_ANY one would need to do a memcpy separate from the variable declaration/initialization, which is less efficient. I've fixed this at least for the address in the latest revision.

Reply:
> If I went with a define similar to BT_ADDR_LE_ANY one would need to do a memcpy separate from the variable declaration/initialization, which is less efficient.

I'm not sure this is correct, BT_ADDR_LE_ANY is a pointer, which BT_IRK_LE_ANY could also be without having to do any copying.

    #define BT_ADDR_LE_ANY  (&(bt_addr_le_t) { 0, { { 0, 0, 0, 0, 0, 0 } } }) // current definition
    #define BT_IRK_LE_ANY   ((u8_t[16]) { 0 })
    ...
    u8_t * irk1   = BT_IRK_LE_ANY; // works
    u8_t irk2[]   = BT_IRK_LE_ANY; // doesn't work (non-constant initializer)
    u8_t irk3[16] = BT_IRK_LE_ANY; // doesn't work (non-constant initializer)
    foo(BT_IRK_LE_ANY);            // works (foo takes u8_t pointer argument)

Reply:
I understand what you mean. I'm not sure we should encourage passing the *_ANY macros directly as input though, since the result is the same as passing NULL, except that the caller will have no way to retrieve the value afterward while still having consumed the memory on the stack to store it. `u8_t *irk = BT_IRK_ANY;` would indeed be a sane thing, with a slightly heavier memory footprint due to pointer (4 bytes) + 16 bytes instead of just 16 bytes. If there's a general consensus that the macro does improve the resulting code, then I'm not opposed to adding it, but it still feels a bit "low value" to me.

Reply:
Sure, we can wait and see if it makes sense.